ci-setup/angular-gitlab.md

```diff
@@ -0,0 +1,236 @@
+
+```yaml
+variables:
+  merge_request_branches: "/^main|development$"
+```
+
+You can modify the varibale as per your needs. For example if you want to run the jobs for merge requests targeting any release branch you can modify the condition as following:
+
+```yaml
+variables:
+  merge_request_branches: "/^main|development|release.*/"
+```
+
+### Adding Rule Templates
+The rules template specifies when a job should run. In this example, jobs will run for merge requests targeting the specified branches in your variable".
+
+```yaml
+.rules_template: &rules_template
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event" && ($CI_MERGE_REQUEST_TARGET_BRANCH_NAME =~ $merge_request_branches)'
+```
+
+### Writing Jobs for Linting and Building
+Create jobs within each stage to perform linting and building tasks.
+
+```yaml
+linting:
+  stage: lint
+  script:
+    - npm install
+    - npm run lint
+  tags:
+    - Node-18-LTS
+  <<: *rules_template
+
+building:
+  stage: build
+  script:
+    - npm install
+    - npm run build
+  dependencies:
+    - linting
+  tags:
+    - Node-18-LTS
+  <<: *rules_template
+```
+
+### Complete `.gitlab-ci.yml` Configuration
+Here is the completed configuration for your .gitlab-ci.yml file:
+
+```yaml
+stages:
+  - lint
+  - build
+
+variables:
+  merge_request_branches: "/^main|development$"
+
+.rules_template: &rules_template
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event" && ($CI_MERGE_REQUEST_TARGET_BRANCH_NAME =~ $merge_request_branches)'
+
+linting:
+  stage: lint
+  script:
+    - npm install
+    - npm run lint
+  tags:
+    - Node-18-LTS
+  <<: *rules_template
+
+building:
+  stage: build
+  script:
+    - npm install
+    - npm run build
+  dependencies:
+    - linting
+  tags:
+    - Node-18-LTS
+  <<: *rules_template
+```
+[Back to top](#table-of-contents)
+## 4. Testing the CI Pipeline
+
+### Creating Merge Requests
+1. Create a new branch in your Angular application repository.
+2. Make changes to your code and push the branch to GitLab.
+3. Create a merge request (MR) targeting the main or development branch.
+
+### Observing Pipeline Execution
+
+1. Navigate to your merge request on GitLab.
+2. Observe the pipeline execution as it runs the linting and building stages.
+3. Check the job logs for any errors or warnings.
+4. Ensure that the pipeline status reflects the success or failure of the linting and building stages.
+
+[Back to top](#table-of-contents)
+## 5. Visualizing the CI Process in GitLab
+In this section, we will visually explain the Continuous Integration (CI) process in GitLab and how it works for your Angular application.
+
+### 1. Developer Creates Merge Request
+When a developer completes a feature or bug fix, they create a new branch in the GitLab repository. They then make changes to the code and create a Merge Request (MR) for code review.
+![Developer Creates Merge Request](assets/angular-developer-creates-mr.png)
```

Comment:
the filename convention for assets should be

`<parent_file_name>_<image_name>.<extension>`

this should be then

`angular-gitlab_angular-developer-creates-mr.png`
- `parent_file_name`: `angular-gitlab`
- `image_name`: `angular-developer-creates-mr`
- `extension`: `png`

Fix elsewhere.

Reply:
Good change to make. Updated the file names as per the comment. Not to repeat preferred this `angular-gitlab_developer-creates-mr.png` instead of  `angular-gitlab_angular-developer-creates-mr.png`. Applied the same for others.